### ISSUE TITLE
Error while saving GPX. The distance calculator plugin can't save GPX route.

### ISSUE THREAD

**Issue by enomat** [osmandapp/Osmand#1667]
Steps to reproduce:
- enable "Distance calculator & planning tool" plugin
- create some points on the map
- tap the ruler then choose "Save as GPX track"
- type something (for GPX file name) then tap "Save"

This error message will appear:
"Error while saving GPX"

<img src="https://cloud.githubusercontent.com/assets/12612250/9304259/7947d552-44e6-11e5-8c76-89c0b2187a19.png" height="500px">

Tested on this device: Samsung Galaxy S2 I9100, Android 4.1.2

Osmand:
- version: 2.2.0#12479M  and the latest, 2.2.0#9061D
- file name: OsmAnd-nightly.apk
- downloaded from here:
http://download.osmand.net/latest-night-build/

I've also tried it on the old Osmand 1.9.5 version, and I get the same error (so it has the same problem).

**Comment by vshcherb** [comment]
It would be good to get more information if possible, using aLogCat for example. The only possible scenario I see that OsmAnd can't write to SD card.

**Comment by enomat** [comment]
Good news. I found what caused the problem. The problem was painfully simple: the "tracks" folder did not exist.

I chose "Open existing GPX" from the ruler icon, then "No GPX files found in the tracks folder" error appeared. This message reminded me that the tracks folder might not exist. I opened the file manager and created the tracks folder, and no error messages appear anymore.

Solution:
Osmand (distance calculator plugin) should create the "tracks" folder if it does not exist when saving/opening a GPX file instead of showing the error message.

(The "Trip recording" plugin automatically creates the "tracks" folder when I choose "Save current GPX track".)
